### DISCUSSION TITLE
Using with grbl

### DISCUSSION THREAD

**Discussion by hackintoshlover** [rdudhagra/Sand-Table#53] [Q&A] [answered]
Hi Ravi,
Wanted to use the device with grbl firmware instead of marlin.
All the codes are almost similar except marlin uses g28 for homing and grbl uses $h.
Where should i make changes in the code so that it parse $h on bootup.
Rigjt now when the raspberry pi boots up and loads the server, my table homes automatically.
So i assume its send g28 for marlin, i want to send $h instead.
Kindly help.

**Comment by rdudhagra** [comment, marked as answer]
Hey! Take a look at https://github.com/rdudhagra/Sand-Table/blob/master/Website/server/routes/gcode_controller.js (Line 352). You'll have to modify that line to match what grbl is looking for (and perhaps the subsequent Marlin-specific commands as well).

**Comment by hackintoshlover** [comment]
Hey Ravi,
Apologies for the late reply.
I tried everything and it runs perfect.
I am currently doing some testing and will post a working video with codes asap.

In the meanwhile i needed some help:
1) How to add a power button icon / physical button for safe shutdown so that sdcard does not get corrupted. (My sd card went bad 5 times within a week due to direct power off)
2) I want to avoid using second arduino for lighting control. How to integrate it directly in the app? In this way the app runs on raspberry pi zero as well.
3) My router likes to automatically change my pi's ip address which results in me checking the sandtables ip address in my router setting always.
I tried changing pi's ip address to static but that creates ip conflict with some other devices if ip address match.
I also tried reserving pi's ip address in my router.
All these makes it a it job.
Can we make it like sandtable.local so that even if the pi's ip address changes, if the user types sandtable.local in a browser it starts the webpage?

**Reply by rdudhagra** [reply]
1 and 2 involve a lot of code changes. I'd look for an example/something online that you can use as a starting point. For #2 check out another project I have on my account called LightBox (beware though it's not documented/messy).

3. That's the point of using the .local address (it should update by itself when the IP changes). It's weird though that setting a static IP with the router isn't working—I'd look into getting that to work first.

**Comment by hackintoshlover** [comment]
Thanks, I'll check for point 2 and try something.
Have you seen dubnom's sandtable software?
The UI is a bit outdated but i think it's epic.
Have a look:
https://github.com/dubnom/sandtable
